test/queues.c

```diff
@@ -2,6 +2,36 @@
 #include "greatest.h"
 #include "queues.h"
 
+TEST test_queue_length(void)
+{
+        queues_init();
+
+        struct notification *n;
+
+        n = test_notification("n1", 0);
+        queues_notification_insert(n);
+        queues_notification_close(n, REASON_UNDEF);
+
+        n = test_notification("n2", 0);
+        queues_notification_insert(n);
+        queues_update(STATUS_NORMAL);
+
+        n = test_notification("n3", 0);
+        queues_notification_insert(n);
+
+        QUEUE_LEN_ALL(1,1,1);
```

Comment:
It'd be a good idea to also check as you're adding the notifications to ensure that they're being put in the correct queue rather than only at the end.

Reply:
It's not important which notification is where. It's just an assertion, that every queue will contain an element.

And it seems to be redundant, but think about it like an assertion. If the `queues_notification_insert` and `queues_update` will fail, the test doesn't step into the problem, that `queues_length_*` reports the wrong size.